validation/validate.py

```diff
@@ -170,17 +170,80 @@ def _score_k_fold(thresholds: np.ndarray,
     return accuracy, recall, precision
 
 
-def _get_target_faces(embeddings1: List[List[float]],
-                      embeddings2: List[List[float]],
+def _get_target_faces(embeddings1: List[FaceVector],
+                      embeddings2: List[FaceVector],
                       distance_metric: DistanceMetric,
-                      is_match: bool) -> Tuple[List[float], List[float]]:
+                      is_match: bool) -> Tuple[FaceVector, FaceVector]:
     X, Y = zip(*[(emb1, emb2) for emb1 in embeddings1 for emb2 in embeddings2])
     distances = _distance_between_embeddings(X, Y, distance_metric)
     distance_criteria = min if is_match else max
```

Comment:
Don't we want to find the best match (min distance) for both matching and unmatching pairs? This would simulate what we would do in real life when we don't know whether the images are matching or not.

Reply:
Wouldn't the following work?

```py
is_match = False
for vector_1, vector_2 in cross_product(embeddings_face_1, embeddings_face_2):
  if distance(vector_1, vector_2) < threshold:
    is_match = True
    break
```

Reply:
Yes, that works if we define a threshold for the distance calculation that determines a match. The approach above does not define a threshold

Reply:
Ahh ok that makes sense to me!

Reply:
How would we like to proceed?